src/components/shared/transactionSummary/index.js

```diff
@@ -10,8 +10,101 @@ import CheckBox from '@toolbox/checkBox';
 import HardwareWalletIllustration from '@toolbox/hardwareWalletIllustration';
 import PassphraseInput from '@toolbox/passphraseInput';
 import Tooltip from '@toolbox/tooltip/tooltip';
+import copyToClipboard from 'copy-to-clipboard';
+import Icon from '@toolbox/icon';
+import { tokenMap } from '@constants';
 import styles from './transactionSummary.css';
 
+const Footer = ({
+  confirmButton, cancelButton, footerClassName, showCancelButton,
+  isHardwareWalletConnected, confirmation, isConfirmed, isMultisignature,
+  account, secondPassphrase, t, transaction, createTransaction,
+}) => {
+  if (isHardwareWalletConnected) {
+    return null;
+  }
+
+  const [download, setDownload] = useState(false);
+  const [copy, setCopy] = useState(false);
+  const [copied, setCopied] = useState(false);
+
+  const onDownload = () => {
+    const anchor = document.createElement('a');
+    anchor.setAttribute('href', `data:text/json;charset=utf-8,${encodeURIComponent(JSON.stringify(transaction))}`);
+    anchor.setAttribute('download', `tx-${transaction.moduleID}-${transaction.assetID}.json`);
+    anchor.click();
+  };
+
+  useEffect(() => {
```

Comment:
We can use callbacks instead of this useEffect. We would be able to remove the `download` and `copy` statuses from the component as well.